### PR TITLE
[Feature][connector-kudu] support Kerberos authentication

### PR DESCRIPTION
**Pull request by NickCodeJourney**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-kudu/pom.xml [hunk 1]
```diff
@@ -56,5 +56,17 @@
...
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client-api</artifactId>
+            <version>3.3.2</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>connector-file-base</artifactId>
+            <version>2.3.2-SNAPSHOT</version>
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-kudu/pom.xml]: use ${project.version} .

File: seatunnel-connectors-v2/connector-kudu/pom.xml [hunk 2]
```diff
@@ -56,5 +56,17 @@
             <artifactId>connector-common</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client-api</artifactId>
+            <version>3.3.2</version>
+            <scope>compile</scope>
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-kudu/pom.xml]: What's the reason you add this dependency?

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/config/KuduSinkConfig.java [hunk 3]
```diff
@@ -51,13 +51,36 @@ public class KuduSinkConfig {
...
+            Options.key("kerberos_principal")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos principal");
+
+    public static final Option<String> KERBEROS_KEYTAB_PATH =
+            Options.key("kerberos_keytab_path")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos keytab file path");
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/config/KuduSinkConfig.java:73]: You can abstract these configuration items into the common module.

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/config/KuduSourceConfig.java [hunk 4]
```diff
@@ -41,4 +41,22 @@ public class KuduSourceConfig implements Serializable {
...
+            Options.key("kerberos_principal")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos principal");
+
+    public static final Option<String> KERBEROS_KEYTAB_PATH =
+            Options.key("kerberos_keytab_path")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("Kudu kerberos keytab file path");
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/config/KuduSourceConfig.java:61]: Same as abrove.

**Review by hailin0 — COMMENTED** [review]
Please update documentation and test cases

**Comment by Carl-Zhou-CN** [comment]
Now that Kudu has added support for Kerberos, if there are no new updates, this PR will be closed.

**Comment by NickCodeJourney** [comment]
agree

**Comment by NickCodeJourney** [comment]
> Now that Kudu has added support for Kerberos, if there are no new updates, this PR will be closed.

agree

**Pull request closed without merging**